mypy/test/testtypes.py

```diff
@@ -621,6 +621,29 @@ def test_type_type(self) -> None:
         self.assert_join(self.fx.type_type, self.fx.type_any, self.fx.type_type)
         self.assert_join(self.fx.type_b, self.fx.anyt, self.fx.anyt)
 
+    def test_literal_type(self) -> None:
+        a = self.fx.a
+        d = self.fx.d
+        lit1 = LiteralType(1, a)
+        lit2 = LiteralType(2, a)
+        lit3 = LiteralType("foo", d)
+
+        self.assert_join(lit1, lit1, lit1)
+        self.assert_join(lit1, a, a)
+        self.assert_join(a, lit1, a)
+        self.assert_join(lit1, lit2, a)
+        self.assert_join(lit1, lit3, self.fx.o)
+        self.assert_join(lit1, self.fx.anyt, self.fx.anyt)
+        self.assert_join(self.fx.anyt, lit1, self.fx.anyt)
+
+        self.assert_simple_join(lit1, lit1, lit1)
+        self.assert_simple_join(lit1, a, a)
+        self.assert_simple_join(a, lit1, a)
+        self.assert_simple_join(lit1, lit2, a)
+        self.assert_simple_join(lit1, lit3, self.fx.o)
+        self.assert_simple_join(lit1, self.fx.anyt, self.fx.anyt)
+        self.assert_simple_join(self.fx.anyt, lit1, self.fx.anyt)
```

Comment:
Test also joins with unions containing literals types.

Reply:
Done. I also didn't see until now that `assert_join` internally already tests symmetry and calls `assert_simple_join`, so I went ahead and removed some existing duplicate tests.